notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/NotificationEventConfig.java

```diff
@@ -0,0 +1,205 @@
+                .notificationEventsTopic(notificationEventsTopic)
+                .bootstrapServers(bootstrapServers)
+                .isKafkaSetupEnabled(isKafkaSetupEnabled)
+                .build();
+    }
+
+    @Data
+    @ConfigurationProperties(prefix = KAFKA_TOPIC_SETUP_PROPERTY)
+    public static class NewTopicProperties {
+        private Integer numPartitions;
+        private Short replicationFactor;
+        private Map<String, String> configs = null;
+
+        public NewTopic buildNewTopic(final String topicName) {
+            // We execute this method and its validations only when 'notification.events.kafka.topic.setup' is true
+            // during KafkaSetupHandler building. If we use @Validated and its constraints, bean loading will fail even
+            // when topic setup is disabled.
+            final String component = "enabled Kafka topic setup";
+
+            Objects.requireNonNull(topicName, getWarningMessageOnNotDefinedProp(component, KAFKA_TOPIC_NAME_PROPERTY));
+            Objects.requireNonNull(numPartitions, getWarningMessageOnNotDefinedProp(component,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".numPartitions")));
+            Objects.requireNonNull(replicationFactor, getWarningMessageOnNotDefinedProp(component,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".replicationFactor")));
+
+            final String gtZeroWarning = " must be greater than zero";
+
+            Preconditions.checkArgument(numPartitions.compareTo(0) > 0,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".numPartitions").concat(gtZeroWarning));
+            Preconditions.checkArgument(replicationFactor.intValue() > 0,
+                    KAFKA_TOPIC_SETUP_PROPERTY.concat(".replicationFactor").concat(gtZeroWarning));
+
+            return new NewTopic(topicName, numPartitions, replicationFactor).configs(configs);
+        }
+    }
+
+    @Data
+    @ConfigurationProperties(prefix = CUSTOM_SCHEMA_TYPE_PREFIX)
+    public static class SchemaParserProperties {
+        private Boolean customEnabled;
+        private String keyParserClass;
+        private String keyParserMethod;
+        private String valueParserClass;
+        private String valueParserMethod;
+
+        public Function<Schema, ?> buildSchemaToKeyRecord() {
+            return Optional.ofNullable(customEnabled)
+                    .filter(Boolean.TRUE::equals)
+                    .map(this::loadKeyParser)
+                    .orElse(NotificationEventUtils::toAvroKeyRecord);
+        }
+
+        public Function<Schema, ?> buildSchemaToValueRecord() {
+            return Optional.ofNullable(customEnabled)
+                    .filter(Boolean.TRUE::equals)
+                    .map(this::loadValueParser)
+                    .orElse(NotificationEventUtils::toAvroValueRecord);
+        }
+
+        private <R extends SpecificRecord> Function<Schema, R> loadKeyParser(Boolean enabled) {
```

Comment:
The `Boolean` parameter here is just a side effect of the `Optional<Boolean>` pattern right? If so I think it's polluting it a bit because `enabled` actually has no effect and so is misleading. Can we maybe rethink this bit?

Reply:
Yes, I define it to traduce it to `this::loadValueParser`, parameter doesn't have an effect. I'll find a better way.

Reply:
Parameter removed